### DISCUSSION TITLE
failed to install spacy v3

### DISCUSSION THREAD

**Discussion by theognis0110** [explosion/spaCy#7439] [Help: Installation] [answered]
Hello,
attempt to install as stated in the documentation after uninstall of the previous version
spacy installs successfully however models installation fails with this error:
    **from murmurhash import hash_unicode
ImportError: cannot import name 'hash_unicode'**

same as the above error occurs with : import spacy

* Operating System: Ubuntu 18.04 LTS
* Python Version Used: python 3.6.6
* spaCy Version Used: the one i get with pip (latest i suppose) 
python -m spacy info --markdown brings the same error as above

i suspect the python version is too low ?
thanks for your help !!

**Comment by theognis0110** [comment]
there's been some improvements:
restarted from a fresh virtualenv based on python 3.9.2
spacy installs well
the english trf model installs well
however french trf model install fails:

Collecting fr-dep-news-trf==3.0.0
  Downloading https://github.com/explosion/spacy-models/releases/download/fr_dep_news_trf-3.0.0/fr_dep_news_trf-3.0.0-py3-none-any.whl (400.1 MB)
     |████████████████████████████████| 400.1 MB 12 kB/s 
Collecting sentencepiece==0.1.91
  Downloading sentencepiece-0.1.91.tar.gz (500 kB)
     |████████████████████████████████| 500 kB 326 kB/s 
    ERROR: Command errored out with exit status 1:
     command: /**********/python -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-qpts4xku/sentencepiece_c6f99e13625743e0975111ebd035a51e/setup.py'"'"'; __file__='"'"'/tmp/pip-install-qpts4xku/sentencepiece_c6f99e13625743e0975111ebd035a51e/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-f90b8cho
         cwd: /tmp/pip-install-qpts4xku/sentencepiece_c6f99e13625743e0975111ebd035a51e/
    Complete output (5 lines):
    Package sentencepiece was not found in the pkg-config search path.
    Perhaps you should add the directory containing `sentencepiece.pc'
    to the PKG_CONFIG_PATH environment variable
    No package 'sentencepiece' found
    Failed to find sentencepiece pkgconfig
    ----------------------------------------
WARNING: Discarding https://files.pythonhosted.org/packages/ba/f6/520b56e5977f62aee48833da8b4ff2fdc2b10ebfa0dd78556b1d707d4086/sentencepiece-0.1.91.tar.gz#sha256=f9700cf607ea064d9fad34c751fbf49953dcc56fe68c54b277481aa0aec5c18f (from https://pypi.org/simple/sentencepiece/). Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
ERROR: Could not find a version that satisfies the requirement sentencepiece==0.1.91 (from fr-dep-news-trf)
ERROR: No matching distribution found for sentencepiece==0.1.91

I can't locate the sentencepiece.pc file so will try with a different french model but i need the trf dependancy accuracy so too bad if i can't use the trf......

**Comment by theognis0110** [comment, marked as answer]
all good under **python 3.8.8** 
now is the fun part....
:)

**Comment by adrianeboyd** [comment]
Thanks for reporting back with your solution!

Generally the original problem with weird import errors is fixed by installing a new venv and the problem with sentencepiece by using python <=3.8.

Python 3.6 should work, but you should be sure to upgrade pip and related tools in your new venv (`pip install -U pip setuptools wheel`) before installing spacy.

The right version of sentencepiece isn't straightforward to install for python 3.9, so it's easiest with python 3.8 at the latest. If you do want to install sentencepiece with python 3.9, see: https://github.com/explosion/spaCy/issues/6969#issuecomment-775117239
